test/compile/compile.test.ts

```diff
@@ -47,6 +47,124 @@ describe('compile/compile', () => {
     expect(spec.marks.length).toEqual(1); // just the root group
   });
 
+  it('should drop fit in top-level properties for discrete x discrete chart', () => {
+    log.wrap(localLogger => {
+      const spec = compile({
+        data: {
+          values: [{x: 'foo', y: 'bar'}]
+        },
+        autosize: 'fit',
+        mark: 'point',
+        encoding: {
+          x: {field: 'x', type: 'nominal'},
+          y: {field: 'y', type: 'nominal'}
+        }
+      }).spec;
+
+      expect(localLogger.warns[0]).toEqual(log.message.droppingFit());
+      expect(spec.autosize).toBe('pad');
+    });
+  });
+
+  it('should drop fit-y in top-level properties for quantitative x discrete chart', () => {
+    log.wrap(localLogger => {
+      const spec = compile({
+        data: {
+          values: [{x: 1, y: 'bar'}]
+        },
+        autosize: 'fit',
+        mark: 'point',
+        encoding: {
+          x: {field: 'x', type: 'quantitative'},
+          y: {field: 'y', type: 'nominal'}
+        }
+      }).spec;
+
+      expect(localLogger.warns[0]).toEqual(log.message.droppingFit('y'));
+      expect(spec.autosize).toBe('fit-x');
+    });
+  });
+
+  it('should drop fit-x in top-level properties for discrete x quantitative chart', () => {
+    log.wrap(localLogger => {
+      const spec = compile({
+        data: {
+          values: [{x: 'foo', y: 1}]
+        },
+        autosize: 'fit',
+        mark: 'point',
+        encoding: {
+          x: {field: 'x', type: 'nominal'},
+          y: {field: 'y', type: 'quantitative'}
+        }
+      }).spec;
+
+      expect(localLogger.warns[0]).toEqual(log.message.droppingFit('x'));
+      expect(spec.autosize).toBe('fit-y');
+    });
+  });
+
+  it('should NOT drop fit in top-level properties for specified width/height chart', () => {
+    log.wrap(localLogger => {
+      const spec = compile({
+        data: {
+          values: [{x: 'foo', y: 'bar'}]
+        },
+        autosize: 'fit',
+        width: 400,
+        height: 400,
+        mark: 'point',
+        encoding: {
+          x: {field: 'x', type: 'nominal'},
+          y: {field: 'y', type: 'nominal'}
+        }
+      }).spec;
+
+      expect(localLogger.warns.length).toEqual(0);
+      expect(spec.autosize).toBe('fit');
+    });
+  });
+
+  it('should NOT drop fit-y in top-level properties for specified height chart', () => {
+    log.wrap(localLogger => {
+      const spec = compile({
+        data: {
+          values: [{x: 1, y: 'bar'}]
+        },
+        height: 400,
+        autosize: 'fit',
+        mark: 'point',
+        encoding: {
+          x: {field: 'x', type: 'quantitative'},
+          y: {field: 'y', type: 'nominal'}
+        }
+      }).spec;
+
+      expect(localLogger.warns.length).toEqual(0);
+      expect(spec.autosize).toBe('fit');
+    });
+  });
+
+  it('should NOT drop fit-x in top-level properties for specified width chart', () => {
```

Comment:
Actually, you don't need this extra level of `() => {`.  

See this removed code https://github.com/vega/vega-lite/pull/5224/files#diff-9885764e87ed89fc726e6fe7d9a5ceeeL127, for example. 

Hopefully this isn't the cause of the low coverage, but maybe it's relevant.

Reply:
fixed in latest commit